unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Interfaces/EntityIdHelper.cs

```diff
@@ -11,25 +11,105 @@ namespace DCL
+                {
+                    case (long)SpecialEntityId.SCENE_ROOT_ENTITY:
+                        return SpecialEntityIdLegacyLiteral.SCENE_ROOT_ENTITY;
+
+                    case (long)SpecialEntityId.FIRST_PERSON_CAMERA_ENTITY_REFERENCE:
+                        return SpecialEntityIdLegacyLiteral.FIRST_PERSON_CAMERA_ENTITY_REFERENCE;
 
-            return entityIdToLegacyId[entityId];
+                    case (long)SpecialEntityId.AVATAR_ENTITY_REFERENCE:
+                        return SpecialEntityIdLegacyLiteral.AVATAR_ENTITY_REFERENCE;
+
+                    case (long)SpecialEntityId.AVATAR_POSITION_REFERENCE:
+                        return SpecialEntityIdLegacyLiteral.AVATAR_POSITION_REFERENCE;
+
+                    case (long)SpecialEntityId.THIRD_PERSON_CAMERA_ENTITY_REFERENCE:
+                        return SpecialEntityIdLegacyLiteral.THIRD_PERSON_CAMERA_ENTITY_REFERENCE;
+                }
+            }
+            
+            // if it has the 0x8000_0000 mask, then it *should* be stored in the entityIdToLegacyId map
+            if ((entityId & 0x80000000) != 0 && entityIdToLegacyId.ContainsKey(entityId)) {
+                // the entity has the 0x80000000 mask signaling it was an invalid ID
+                return entityIdToLegacyId[entityId];
+            }
+
+            // lastly, fallback to generating the entityId based on the number.
+            return ToBase36EntityId(entityId);
         }
+        
+        private static string ToBase36EntityId(long value)
+        {
+            // TODO(mendez): the string builder approach can be better. It does use more allocations
+            //               than the optimal. The ideal scenario would use a `stackalloc char[13]` instead 
+            //               of StringBuilder but my knowledge and tools for C# are limited at the moment
+            const string base36 = "0123456789abcdefghijklmnopqrstuvwxyz";
+            var sb = new StringBuilder(13);
 
+            value = value >> 9; // recover from bit-shift
+            value -= 1; // recover the added 1
+
+            do
+            {
+                sb.Insert(0, base36[(byte)(value % 36)]);
+                value /= 36;
+            } while (value != 0);
+
+            // E prefix for entities
+            sb.Insert(0, "E");
+
+	          return sb.ToString();
+        }
+        
         public long EntityFromLegacyEntityString(string entityId)
         {
+            long entityIdLong = GetConvertedEntityId(entityId);
+            return entityIdLong;
+        }
+
+        long GetConvertedEntityId(string entityId) {
+            // entity Ids in base36 start with E
+            if (entityId[0] == 'E')
+            {
+                long result = 1;
+                long power = 1;
+                for (var i = entityId.Length - 1; i > 0; i--)
+                {
+                    char charCode = entityId[i];
+                    bool isBase36Number = charCode >= '0' && charCode <= '9';
```

Comment:
simple suggestion, 🤔  up to you
```suggestion
                    bool isNumber = charCode >= '0' && charCode <= '9';
```